my_enumerable.rb

```diff
@@ -1,92 +1,178 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength
 module Enumerable
   def my_each
-    return to_enum(_method_) unless block_given?
+    return to_enum(__method__) unless block_given?
 
+    array = to_a
     i = 0
-    until i == length
-      yield(self[i])
+    until i == array.length
+      yield(array[i])
       i += 1
     end
+    if array.is_a?(Numeric) && Range.new(array.first, array.last) == self
+      self
+    else
+      array
+    end
```

Comment:
- [ ] When a block is given and my_each is called on a range, it should return the range itself after calling the given block once for each element.

```ruby
block = proc { |num| num < 4 }
range = Range.new(5, 50) 
range.my_each(&block) === range.each(&block) => false
```

* I can see that you have previously solved this issue however it is failing this test case again. Take a closer look at what you have done since solving it to see what caused it to break.